### ISSUE TITLE
Unable to set log_level

### ISSUE THREAD

**Issue by igolman** [chef-cookbooks/docker#999]
### Cookbook version
4.2.0

### Chef-client version
12.19.36

### Platform Details
Ubuntu 16.04

### Scenario:
set log_level for docker_service

### Steps to Reproduce:
in _my_docker::default_ recipe:
```
docker_service 'default' do
  node['my_docker']['service_config'].each do |method_name, value|
    send(method_name, value)
  end
end
```
Then define _docker_service_ via Chef role:
```
my_docker: {
    service_config: {
      log_level: :error
}
```
or in _kitchen.yml_
```
  - name: default
    run_list:
      - recipe[my_docker::default]
    attributes:
      my_docker:
        service_config:
          log_level: :error
```

### Expected Result:
chef-client runs flawlessly, log_level set to :error 

### Actual Result:
```
       Recipe Compile Error in /tmp/kitchen/cache/cookbooks/my_docker/recipes/default.rb
       =============================================================================

       Chef::Exceptions::ValidationFailed
       ----------------------------------
       Property log_level must be one of: :debug, :info, :warn, :error, :fatal, nil!  You passed ":error".

       Cookbook Trace:
       ---------------
         /tmp/kitchen/cache/cookbooks/my_docker/recipes/default.rb:3:in `block (2 levels) in from_file'
         /tmp/kitchen/cache/cookbooks/my_docker/recipes/default.rb:2:in `each'
         /tmp/kitchen/cache/cookbooks/my_docker/recipes/default.rb:2:in `block in from_file'
         /tmp/kitchen/cache/cookbooks/my_docker/recipes/default.rb:1:in `from_file'
         /tmp/kitchen/cache/cookbooks/my_docker/recipes/default.rb:1:in `from_file'

       Relevant File Content:
       ----------------------
       /tmp/kitchen/cache/cookbooks/docker_dkd/recipes/docker_install.rb:

         1:  docker_service 'default' do
         2:    node['my_docker']['service_config'].each do |method_name, value|
         3>>     send(method_name, value)
         4:    end
         5:  end
         6:
```

**Comment by chasebolt** [comment]
can you try with "error" instead of ":error" and see if that has a
different result?

**Comment by igolman** [comment]
> can you try with "error" instead of ":error" and see if that has a different result?

Sure I tried,  same result..
  `    [2018-06-19T17:01:57+00:00] ERROR: Property log_level must be one of: :debug, :info, :warn, :error, :fatal, nil!  You passed "error".`

Looks to me, like the elements of :log_level property can never be matched, since Chef-Client/Kitchen convert symbol attributes from  to strings...
https://github.com/chef-cookbooks/docker/blob/master/libraries/docker_service_base.rb
`property :log_level, [:debug, :info, :warn, :error, :fatal, nil]`

However passing the property explicitly via cookbooks recipe works... 
```
docker_service 'default' do
  ..
  ..
  log_level :error
end
```

===

I've found a few bug reports regarding the mentioned above Client/Kitchen problem:
1. https://github.com/test-kitchen/test-kitchen/pull/652
2. https://tickets.opscode.com/browse/COOK-4736

===

Since the :log_level is only for 'syslog' log_driver relevant and the log_driver is already typed as a String, I would  suggest to change the :log_level property to String as well..
https://docs.docker.com/config/containers/logging/syslog/

pull request: [#1000](https://github.com/chef-cookbooks/docker/pull/1000/)

Regards,

ig.

**Comment by dheerajd-msys** [comment]
Closing as it is fixed in #1000
